### PR TITLE
#351 run js functions in their own goroutine

### PR DESCRIPTION
**Pull request by absorbb**

### CODE REVIEW

File: server/templates/template_executor.go [hunk 1]
```diff
@@ -42,16 +49,33 @@ func (gte *goTemplateExecutor) Format() string {
...
+
+type asyncJsTemplateExecutor struct {
+	incoming chan events.Event
+	results chan interface{}
+	transformedExpression string
+	loadingError error
 }
 
-func newJsTemplateExecutor(expression string, extraFunctions template.FuncMap) (*jsTemplateExecutor, error) {
+type pooledTemplateExecutor struct {
```
Comment (xtreding) [inline server/templates/template_executor.go:73]: is it used anywhere? If not, we shouldn't keep it in the codebase

File: server/templates/template_executor.go [hunk 2]
```diff
@@ -63,12 +87,70 @@ func newJsTemplateExecutor(expression string, extraFunctions template.FuncMap) (
...
+
+
+func (jte *asyncJsTemplateExecutor) start(extraFunctions template.FuncMap) {
+	//loads javascript into new vm instance
+	function, err := LoadTemplateScript(jte.transformedExpression, extraFunctions)
 	if err != nil {
-		return nil, fmt.Errorf("js loading error: %v", err)
+		jte.loadingError =  fmt.Errorf("%s: %v\ntransformed function:\n%v\n",jsLoadingErrorText, err, jte.transformedExpression)
+	}
+	for {
```
Comment (xtreding) [inline server/templates/template_executor.go:108]: For such cases we have `safego` package that contains `safego.RunWithRestart()` func. Please take a look at the example https://github.com/jitsucom/jitsu/blob/master/server/adapters/http.go#L98

**Review by xtreding — APPROVED** [review]

File: server/templates/template_executor.go (hunk 1 above)
Comment (absorbb) [reply server/templates/template_executor.go:73]: not used. will deleted it

File: server/templates/template_executor.go (hunk 2 above)
Comment (absorbb) [reply server/templates/template_executor.go:98]: done. but for entire start() function.

**Pull request merged**